flybywiresim/aircraft
# fix(a380x): Port Over the New A32NX GPWS to the A380X

Fixes #8802 

## Summary of Changes

Removed the option for a landing flaps 3 configuration.
Re-did the a380x's GPWS mode 4 logic and replaced it with new mode 4 logic. 
Added in various different fixes to the GPWS logic from the new GPWS update.

## Screenshots (if necessary)



## References
![image](https://github.com/user-attachments/assets/a17e4077-fe0d-4561-9abf-062e968a52c9)
![image](https://github.com/user-attachments/assets/72a53c2e-3bef-492b-9ecb-07a5f7b9415c)


Discord username (if different from GitHub):
Capt. Crystal

## Testing instructions
1. Load into an approach in the A380, any airport should work.
2. Confirm that the necessary GPWS callouts are announced at the correct altitude AGL. I recommend using flight recorder off of Flightsim.to so you can pause when you hear the callout to verify that it was made at the correct altitude AGL.
3. End flight
4. Load into the A380 in a mountainous area. 
5. Confirm that the correct GPWS terrain callouts are made. The "correct" callouts are shown in the Reference section.


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
